lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -81,16 +81,18 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         def serializable_hash_for_collection(options)
           hash = { data: [] }
+          included = []
           serializer.each do |s|
             result = self.class.new(s, instance_options.merge(fieldset: fieldset)).serializable_hash(options)
             hash[:data] << result[:data]
+            next unless result[:included]
 
-            if result[:included]
-              hash[:included] ||= []
-              hash[:included] |= result[:included]
-            end
+            included |= result[:included]
           end
 
+          included.delete_if { |resource| hash[:data].include?(resource) }
```

Comment:
deletes resource from `included` if is a root resource

this included stuff is getting messy. needs a refactor before it sends even more vines through the class